### PR TITLE
Handshake expiration date

### PR DESCRIPTION
**Pull request by mjoyce91**
- Adds `expiration_date` to the BidHandshake model, which is now the only field that can be updated through the request `body`
- `expiration_date` is reset to null when handshake offers are revoked 
- `hs_date_expiration` added as new prop returned by `get_bidder_handshake_data` service

### CODE REVIEW

File: talentmap_api/bidding/views/bidhandshake.py [hunk 1]
```diff
@@ -87,7 +112,7 @@ def delete(self, request, pk, cp_id, format=None):
         else:
             user = UserProfile.objects.get(user=self.request.user)
             hs.update(last_editing_user=user, bidder_perdet=pk, cp_id=cp_id, status='R',
-                update_date=datetime.now(), date_revoked=datetime.now())
+                update_date=datetime.now(), date_revoked=datetime.now(), expiration_date=None)
```
Comment (scott062) [inline talentmap_api/bidding/views/bidhandshake.py:115]: I can't think of any use-case yet, but do we want to keep the original `expiration_date` until they re-offer (where you reset it above)? 

I'm thinking this maintains the "history" of the current HS until re-offer where everything gets reset.

File: talentmap_api/bidding/serializers.py [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
+# Only use serializer for PUT body data
+class BidHandshakeOfferSerializer(PrefetchedSerializer):
+
+    def validate(self, data):
+        # The keys can be missing in partial updates
+        if "expiration_date" in data:
+            if data["expiration_date"] < maya.now().datetime():
+                raise serializers.ValidationError({
+                    "expiration_date": "Expiration date cannot be earlier than current date (Eastern Time).",
+                })
```
Comment (scott062) [inline talentmap_api/bidding/serializers.py:24]: Is it simple enough to setup future time rules here? Say we implement a minimum 8 HR rule, where you can't set an offer expiration within 8 hrs of the current time - pending future business rules. Or do we have to do weird manipulation of the datetime object?

**Comment by scott062** [comment]
Everything works as intended on swagger - just a couple comments but otherwise good to go 👍 . Testing FE integration now

File: talentmap_api/bidding/serializers.py (hunk 2 above)
Comment (mjoyce91) [reply talentmap_api/bidding/serializers.py:24]: yeah we could definitely do that here

File: talentmap_api/bidding/views/bidhandshake.py (hunk 1 above)
Comment (mjoyce91) [reply talentmap_api/bidding/views/bidhandshake.py:115]: I was thinking about that too. I don't see why not

Comment (mjoyce91) [reply talentmap_api/bidding/views/bidhandshake.py:115]: Taking a look at this - trying to set expiration_date as an optional field

Comment (mjoyce91) [reply talentmap_api/bidding/views/bidhandshake.py]: Updated

**Review by scott062 — APPROVED** [review]

**Pull request merged**